src/bin/pg_autoctl/keeper.c

```diff
@@ -2100,12 +2122,18 @@ keeper_reload_configuration(Keeper *keeper, bool firstLoop)
 void
 keeper_call_reload_hooks(Keeper *keeper, bool firstLoop)
 {
+	int errors = 0;
 	for (int index = 0; KeeperReloadHooks[index]; index++)
 	{
-		/* at the moment we ignore the return values from the reload hooks */
-		(void) (*KeeperReloadHooks[index])(keeper, firstLoop);
+		if (!(*KeeperReloadHooks[index])(keeper, firstLoop))
+		{
+			errors++;
+		}
 	}
 
-	/* we're done reloading now. */
-	asked_to_reload = 0;
+	/*
+	 * If there are no errors reloading the configuration, we are done.
+	 * Otherwise, we will have to retry the reload.
+	 */
+	asked_to_reload = (errors > 0);
```

Comment:
I'm not fond of eternal reload retries, maybe we need a counter and a MaxR/MaxT strategy as in other places. That said, for a first PR, we may be able to accept a reload-forever policy here. I'm worried about picking a password that we would always fail to SET. Is that possible?

Reply:
> I'm worried about picking a password that we would always fail to SET. Is that possible?

Yes, if for example the [`passwordcheck` contrib module](https://www.postgresql.org/docs/13/passwordcheck.html) is installed and doesn't like the password.

Reply:
See another comment in this PR for an alternative approach that allows removing the reload-again-and-again hack, which I think is a terrible idea now that I realise that the first thing we do is `keeper_config_update` which calls `keeper_config_write_file`. That would prevent users from doing any other edits to the configuration file, or they would need to time it perfectly with the reload signal.